tests/components/dynalite/test_init.py

```diff
@@ -79,6 +82,114 @@ async def test_async_setup(hass):
+        mock_req_area_pres.reset_mock()
+        await hass.services.async_call(
+            dynalite.DOMAIN, "request_area_preset", {"host": "5.6.7.8", "area": 4},
+        )
+        await hass.async_block_till_done()
+        mock_req_area_pres.assert_called_once_with(4, 1)
+        mock_req_area_pres.reset_mock()
+        await hass.services.async_call(
+            dynalite.DOMAIN, "request_area_preset", {"host": "6.5.4.3", "area": 5},
+        )
+        await hass.async_block_till_done()
+        mock_req_area_pres.assert_not_called()
+        mock_req_area_pres.reset_mock()
+        await hass.services.async_call(
+            dynalite.DOMAIN,
+            "request_area_preset",
+            {"host": "1.2.3.4", "area": 6, "channel": 9},
+        )
+        await hass.async_block_till_done()
+        mock_req_area_pres.assert_called_once_with(6, 9)
+        mock_req_area_pres.reset_mock()
+        await hass.services.async_call(
+            dynalite.DOMAIN, "request_area_preset", {"host": "1.2.3.4", "area": 7},
+        )
+        await hass.async_block_till_done()
+        mock_req_area_pres.assert_called_once_with(7, 1)
+
+
+async def test_service_request_channel_level(hass):
+    """Test requesting the level of a channel via service call."""
+    with patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.async_setup",
+        return_value=True,
+    ), patch(
+        "dynalite_devices_lib.dynalite.Dynalite.request_channel_level",
+        return_value=True,
+    ) as mock_req_chan_lvl:
+        assert await async_setup_component(
+            hass,
+            dynalite.DOMAIN,
+            {
+                dynalite.DOMAIN: {
+                    dynalite.CONF_BRIDGES: [
+                        {
+                            CONF_HOST: "1.2.3.4",
+                            dynalite.CONF_AREA: {"7": {CONF_NAME: "test"}},
+                        },
+                        {CONF_HOST: "5.6.7.8"},
+                    ]
+                }
+            },
+        )
+        await hass.async_block_till_done()
+        assert len(hass.config_entries.async_entries(dynalite.DOMAIN)) == 2
+        await hass.services.async_call(
+            dynalite.DOMAIN,
+            "request_channel_level",
+            {"host": "1.2.3.4", "area": 2, "channel": 3},
+        )
+        await hass.async_block_till_done()
+        mock_req_chan_lvl.assert_called_once_with(2, 3)
+        mock_req_chan_lvl.reset_mock()
+        with pytest.raises(MultipleInvalid):
+            await hass.services.async_call(
+                dynalite.DOMAIN, "request_channel_level", {"area": 3},
+            )
+        await hass.async_block_till_done()
+        mock_req_chan_lvl.assert_not_called()
+        await hass.services.async_call(
+            dynalite.DOMAIN, "request_channel_level", {"area": 4, "channel": 5},
+        )
+        await hass.async_block_till_done()
+        assert mock_req_chan_lvl.mock_calls == [call(4, 5), call(4, 5)]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)